### PR TITLE
osd OS4 cluster creation pipeline added

### PR DESCRIPTION
**Pull request by pawelpaszki**
## What
https://issues.jboss.org/browse/INTLY-3942 - osd OS4 cluster creation pipeline added

## Verification
Check the code and the output of this pipeline: https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/ppaszki-create-osd-os4/19/console

### CODE REVIEW

File: jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                }
+            } // stage
+            
+            stage("Begin cluster creation") {
+                rootWorkspaceDir = sh (
+                    script: "pwd",
+                    returnStdout: true
+                ).trim()
+                installConfigDirectory = rootWorkspaceDir + "/rhmi-jenkins/" + clusterInfoDirectory + "/"
+                sh "ocm post /api/clusters_mgmt/v1/clusters --body=${installConfigDirectory}cluster.json > ${clusterInfoFileName}"
```
Comment (tremes) [inline jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: Maybe the `/api/clusters_mgmt/v1/clusters` is a good candidate for a constant.

File: jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+    }}} // timeout, ansiColor, timestamps
+} // node
+
+def isClusterReady(clusterID) {
+    String clusterStatus = sh (
+        script: "ocm get /api/clusters_mgmt/v1/clusters/${clusterID}/status | jq -r .state",
+        returnStdout: true
+    ).trim()
+    return clusterStatus.contains("ready")
```
Comment (tremes) [inline jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: Just out of curiosity - what does it return when it's not ready?

File: jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile (hunk 1 above)
Comment (pawelpaszki) [reply jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: sure. makes sense

File: jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile (hunk 2 above)
Comment (pawelpaszki) [reply jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: `installing`

Comment (tremes) [reply jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: OK thanks

File: jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile (hunk 1 above)
Comment (pawelpaszki) [reply jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile]: @tremes I think it is ok now, but there seems to be some issue with the cluster creation infrastructure, therefore the cluster was not provisioned within 2 hours (still provisioning), while it usually takes approx 30 minutes

File: jobs/integr8ly/ocp4/osd/cluster/create/osd-cluster-create.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    parameters:
+      - string:
+          name: 'CLUSTER_NAME'
+          default: ''
+          description: 'Name of the cluster (e.g. <your kerberos username>-small)'
+      - choice:
+          name: 'OCM_TOKEN'
+          choices:
+            - 'ppaszki-ocm-token'
+          description: 'ocm token used to create a cluster. Add your own one to Jenkins credentials and to the list of the choices for this paramter, if not already here (more info here: https://github.com/RHCloudServices/integreatly-help/blob/master/guides/ocp4/osd_dev_cluster.md)'
```
Comment (tremes) [inline jobs/integr8ly/ocp4/osd/cluster/create/osd-cluster-create.yaml]: `paramter` typo

**Comment by tremes** [comment]
I just did a basic review since I am not familiar with particular steps used in this pipeline.

**Comment by pawelpaszki** [comment]
Thanks @tremes - here is the doc for more info of the process: https://github.com/RHCloudServices/integreatly-help/blob/master/guides/ocp4/osd_dev_cluster.md

**Comment by pawelpaszki** [comment]
Pipeline running here: https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/ppaszki-create-osd-os4/21/console

**Comment by tonyxrmdavidson** [comment]
The pipeline finished successfully.   https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/ppaszki-create-osd-os4/21/console

**Pull request merged**